hybrasyl/Objects/Monster.cs

```diff
@@ -627,19 +266,23 @@ public void AssailAttack(Direction direction, Creature target = null)
                 //try to get the creature we're facing and set it as the target.
             }
             
-             Attack(direction, target);
-                
-            
+            // A monster's assail is just a straight attack, no skills involved.
+            SimpleAttack(target);
+                            
             //animation handled here as to not repeatedly send assails.
             var assail = new ServerPacketStructures.PlayerAnimation() { Animation = 1, Speed = 20, UserId = this.Id };
-            var sound = new ServerPacketStructures.PlaySound() { Sound = (byte)1 };
             //Enqueue(assail.Packet());
             //Enqueue(sound.Packet());
             SendAnimation(assail.Packet());
-            PlaySound(sound.Packet());
+            PlaySound(1);
```

Comment:
using the structure is proper for consistency, even if it is only a single byte returned.

Reply:
IMO the packet structure(s) should be used in the `Play*` or `Show*` functions themselves, not in individual functions (I changed the signature to take a numeric ID, as it did before). I don't believe that individual functions and handlers should need to construct packets themselves.

There is a lot of packet structure cleanup work that should be done, but I agree, using the structure, when it finally comes time to put something on the wire, is the correct pattern.